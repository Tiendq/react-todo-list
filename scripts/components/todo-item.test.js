import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { fireEvent } from '@testing-library/react'
import TestRenderer from 'react-test-renderer';
import TodoItem from './todo-item';

describe('TodoItem', () => {
  let container = null;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test('should render correctly', () => {
    let renderedTree = TestRenderer.create(
      <div>
        <TodoItem text="Completed" completed={true} onToggleItem={() => {}} onRemoveItem={() => {}} />
        <TodoItem text="Not completed" completed={false} onToggleItem={() => {}} onRemoveItem={() => {}} />
      </div>
    );

    expect(renderedTree.toJSON()).toMatchSnapshot();
  });

  test('should be 1 completed item', () => {
    render(<TodoItem text="Completed" completed={true} onToggleItem={() => {}} onRemoveItem={() => {}} />, container);
    expect(container.querySelector('.form-check-input').checked).toBe(true);
  });

  test('should mark an item to be completed', () => {
    let onToggleItem = jest.fn();

    render(<TodoItem text="Open item" completed={false} onToggleItem={onToggleItem} onRemoveItem={() => {}} />, container);

    let checkbox = container.querySelector('.form-check-input');
    fireEvent.click(checkbox);

    expect(onToggleItem).toBeCalledTimes(1);
  });

  test('should remove an item', () => {
    let onRemoveItem = jest.fn();

    render(<TodoItem text="Open item" completed={false} onToggleItem={() => {}} onRemoveItem={onRemoveItem} />, container);
    fireEvent.click(container.querySelector('.btn-danger'));

    expect(onRemoveItem).toBeCalledTimes(1);
  });
});
