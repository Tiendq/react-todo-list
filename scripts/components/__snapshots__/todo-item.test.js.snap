// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render correctly 1`] = `
<div>
  <li
    className="form-check todo-item done"
  >
    <label
      className="form-check-label"
    >
      <input
        checked={true}
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
      />
       
      Completed
    </label>
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </li>
  <li
    className="form-check todo-item open"
  >
    <label
      className="form-check-label"
    >
      <input
        checked={false}
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
      />
       
      Not completed
    </label>
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </li>
</div>
`;
